libs/hdf-converters/src/xccdf-results-mapper.ts

```diff
@@ -151,11 +160,18 @@ function extractCci(input: IIdent | IIdent[]): string[] {
 }
 
 function nistTag(input: IIdent | IIdent[]): string[] {
-  const identifiers: string[] = extractCci(input);
   return CCI_NIST_MAPPING.nistFilter(
-    identifiers,
+    extractCci(input),
     DEFAULT_STATIC_CODE_ANALYSIS_NIST_TAGS,
     false
+  ).concat(
+    asArray(input)
+      .filter((x) => !!x)
+      .map((x) => x.text)
+      .map(parse_nist)
+      .filter((x) => !!x)
+      .filter(is_control)
```

Comment:
seems like your sample is not updated properly still?  or this function is borked lol

xccdf-scc-rhel7-hdf.json

![image](https://github.com/mitre/heimdall2/assets/2015821/6345d9a5-84fc-4eef-8a1a-c80120c776ff)

seems like the nist tag array is including the stig control ids, which it shouldn't do

Reply:
`is_control(parse_nist('SV-86515'))` return true... so as far as inspectjs is concerned, 'SV-86515' is a nist control and therefore should be included.

Am I misunderstanding `nist.js`? How should I determine if an item should be included or excluded from this list?

Reply:
this feels like a failure on inspecjs' part in not removing things that are clearly wrong. parse_nist should throw errors/explicitly fail when they are not nist controls/control families. it not doing so confuses the type system by mistakenly typing that string. is_control more or less is only used to narrow down types and doesn't do much actual validation.

anyways as shown there and in 'raw_nist.ts', nist controls look like 'two_letter_family-small_number possibly followed by letters and numbers in parentheses' whereas stig ids are 'S?V-5__or_6_digit_number'.

the issue that arose here is that it thinks 'sv' is a control family and then processes the numbers after it as like 'control "big number" from within that control family' but doesn't take into account the reality of the situation where 'sv' is not a valid control family nor is there a control with an index as high as '86515' in existence.

looking briefly, it seems like the regexes that parse_nist uses are vastly broader than the actually allowed set of inputs. i will experiment to see if it's possible to restrict it to allowed control families.